ui/src/dropdown.tsx

```diff
@@ -118,12 +123,24 @@ const
         if (!selection) return
 
         selection.clear()
-        setSelectedOptions([])
+        setMultiValues([])
         wave.args[name] = []
 
         onChange()
       }
 
+    React.useEffect(() => {
+      if (!isMultivalued) return
```

Comment:
Would this work if `isMultivalued` changes? It's not in the dep array. Checking `values` for `undefined` seems like a safer way here.

Reply:
TBH, I was not even considering that we want to allow to dynamically switch between single and multi, but it worked by simply adding `isMultivalued` to dep array. One caveat: `values` needs to be set to `None` to switch from multi to single.

Repro code

```py
from h2o_wave import app, main, Q, ui


@app("/demo")
async def serve(q: Q) -> None:
    if not q.client.initialized:
        q.client.initialized = True
        q.page["example"] = ui.form_card(
            box="1 1 4 6",
            items=[
                ui.dropdown(
                    name='dropdown',
                    label='dropdown',
                    choices=[
                        ui.choice(name="1", label="Choice 1"),
                        ui.choice(name="2", label="Choice 2"),
                        ui.choice(name="3", label="Choice 3"),
                    ],
                    value="1",
                ),
                ui.button(name="single", label="single (value = 2)"),
                ui.button(name="multi", label="multi (values = [2, 3])"),
                ui.button(name="popup_on", label="popup on"),
                ui.button(name="popup_off", label="popup off"),
            ],
        )
    if q.args.single:
        q.page["example"].items[0].dropdown.value = '2'
        q.page["example"].items[0].dropdown.values = None
    if q.args.multi:
        q.page["example"].items[0].dropdown.values = ['2', '3']
    if q.args.popup_on:
        q.page["example"].items[0].dropdown.popup = 'always'
    if q.args.popup_off:
        q.page["example"].items[0].dropdown.popup = 'never'

    await q.page.save()

```

Also, checked whether changing `popup` dynamically it would work but found some bugs.